crates/sovereign/sovereign-integration-tests/src/impls/build_rollup_driver.rs

```diff
@@ -40,20 +40,44 @@ where
     ) -> Result<SovereignRollupDriver, Bootstrap::Error> {
         let rpc_config = &node_config.runner.rpc_config;
 
-        let rpc_client = HttpClientBuilder::default()
-            .build(format!(
+        let instant = std::time::Instant::now();
+
+        let rpc_client = loop {
```

Comment:
I think we only need a `for _ in 0..10` here as we don't need a precise timeout.

Reply:
I tried to use `for _ in 0..10`, but I realized it breaks the ergonomics, as `for` can't `break` with value.

Hope, my latest commit works.

Reply:
I created a `build_subscription_client` function to help build the client with retry. Also the constructor `HttpClientBuilder` is not async and therefore don't connect to the server on construction, so we can skip the retry there.